simple-loadflow/src/main/java/com/powsybl/loadflow/simple/ac/NewtonRaphson.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2019, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.loadflow.simple.ac;
+
+import com.powsybl.loadflow.simple.equations.EquationSystem;
+import com.powsybl.loadflow.simple.equations.EquationContext;
+import com.powsybl.loadflow.simple.equations.Vectors;
+import com.powsybl.loadflow.simple.network.NetworkContext;
+import com.powsybl.math.matrix.LUDecomposition;
+import com.powsybl.math.matrix.Matrix;
+import com.powsybl.math.matrix.MatrixFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+public class NewtonRaphson {
```

Comment:
I think at some point it could be in a dedicated math package, as it's not dependent on load flow application.

Reply:
Almost independent but equations and variables have an enum type which is specific to power flow. I could use a String instead to have a generic algo?